aries_cloudagent/protocols/issue_credential/v2_0/formats/vc_di/handler.py

```diff
@@ -0,0 +1,458 @@
+
+    def get_format_data(self, message_type: str, data: dict) -> CredFormatAttachment:
+        """Get credential format and attachment objects for use in cred ex messages.
+
+        Returns a tuple of both credential format and attachment decorator for use
+        in credential exchange messages. It looks up the correct format identifier and
+        encodes the data as a base64 attachment.
+
+        Args:
+            message_type (str): The message type for which to return the cred format.
+                Should be one of the message types defined in the message types file
+            data (dict): The data to include in the attach decorator
+
+        Returns:
+            CredFormatAttachment: Credential format and attachment data objects
+
+        """
+        return (
+            V20CredFormat(
+                attach_id=VCDICredFormatHandler.format.api,
+                format_=self.get_format_identifier(message_type),
+            ),
+            AttachDecorator.data_base64(data, ident=VCDICredFormatHandler.format.api),
+        )
+
+    async def _match_sent_cred_def_id(self, tag_query: Mapping[str, str]) -> str:
+        """Return most recent matching id of cred def that agent sent to ledger."""
+
+        async with self.profile.session() as session:
+            storage = session.inject(BaseStorage)
+            found = await storage.find_all_records(
+                type_filter=CRED_DEF_SENT_RECORD_TYPE, tag_query=tag_query
+            )
+        if not found:
+            raise V20CredFormatError(
+                f"Issuer has no operable cred def for proposal spec {tag_query}"
+            )
+        return max(found, key=lambda r: int(r.tags["epoch"])).tags["cred_def_id"]
+
+    async def create_proposal(
+        self, cred_ex_record: V20CredExRecord, proposal_data: Mapping[str, str]
+    ) -> Tuple[V20CredFormat, AttachDecorator]:
+        """Create vc_di credential proposal."""
+        if proposal_data is None:
+            proposal_data = {}
+
+        return self.get_format_data(CRED_20_PROPOSAL, proposal_data)
+
+    async def receive_proposal(
+        self, cred_ex_record: V20CredExRecord, cred_proposal_message: V20CredProposal
+    ) -> None:
+        """Receive vcdi credential proposal.
+
+        No custom handling is required for this step.
+        """
+
+    async def create_offer(
+        self, cred_proposal_message: V20CredProposal
+    ) -> CredFormatAttachment:
+        """Create vcdi credential offer."""
+
+        issuer = AnonCredsIssuer(self.profile)
+        ledger = self.profile.inject(BaseLedger)
+        cache = self.profile.inject_or(BaseCache)
+
+        cred_def_id = await issuer.match_created_credential_definitions(
+            **cred_proposal_message.attachment(VCDICredFormatHandler.format)
+        )
+
+        async def _create():
+            # TODO - implement a separate create_credential_offer for vcdi
```

Comment:
@ianco any clue here quite welcome - do we really need a separate create_credential_offer?

Or can we get away with some kind of custom `issuer` that contains the right schema_id?

Reply:
@tra371 @sarthakvijayvergiya  is it possible that here we should call create_credential_vc_di from https://github.com/hyperledger/aries-cloudagent-python/blob/22d11e2c0084ae8ced2d920714e248abd7ff4f0a/aries_cloudagent/anoncreds/issuer.py#L636 ?

Reply:
but that creates the credential itself, so maybe that is the attachment?